geniza/common/models.py

```diff
@@ -60,3 +71,30 @@ class UserProfile(models.Model):
     def __str__(self):
         # needed for display label in admin
         return "User profile for %s" % (self.user)
+
+
+class DisplayLabelMixin:
+    """Mixin for models with translatable display labels that may differ from full names."""
```

Comment:
not sure I totally follow this in the abstract; could you add a brief example here? is this basically just managing the field vs en_field ? and is it mainly used in django admin?